sklearn/neighbors/_nearest_centroid.py

```diff
@@ -94,6 +97,11 @@ class NearestCentroid(ClassifierMixin, BaseEstimator):
     [1]
     """
 
+    _parameter_constraints = {
+        "metric": [StrOptions(set(_VALID_METRICS)), callable],
```

Comment:
Actually there is a bug in the existing code base as described here #23890. I think this would be a good opportunity to fix it and improve the tests accordingly.

```suggestion
        "metric": StrOptions({"euclidean", "manhattan"}),
```